util/tracecpu/collector_test.go

```diff
@@ -0,0 +1,257 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracecpu
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/gogo/protobuf/types"
+	. "github.com/pingcap/check"
+	"github.com/pingcap/tipb/go-tipb"
+	"google.golang.org/grpc"
+)
+
+const (
+	maxSQLNum = 5000
+)
+
+func TestT(t *testing.T) {
+	TestingT(t)
+}
+
+var _ = Suite(&testTopSQLCollector{})
+
+type testTopSQLCollector struct{}
+
+func (s *testTopSQLCollector) SetUpSuite(c *C) {}
+
+func (s *testTopSQLCollector) SetUpTest(c *C) {}
+
+func testPlanBinaryDecoderFunc(plan string) (string, error) {
+	return plan, nil
+}
+
+func populateCache(ts *TopSQLCollector, begin, end int, timestamp uint64) {
+	// register normalized sql
+	for i := begin; i < end; i++ {
+		key := "sqlDigest" + strconv.Itoa(i+1)
+		value := "sqlNormalized" + strconv.Itoa(i+1)
+		ts.RegisterNormalizedSQL(key, value)
+	}
+	// register normalized plan
+	for i := begin; i < end; i++ {
+		key := "planDigest" + strconv.Itoa(i+1)
+		value := "planNormalized" + strconv.Itoa(i+1)
+		ts.RegisterNormalizedPlan(key, value)
+	}
+	// collect
+	var records []TopSQLRecord
+	for i := begin; i < end; i++ {
+		records = append(records, TopSQLRecord{
+			SQLDigest:  "sqlDigest" + strconv.Itoa(i+1),
+			PlanDigest: "planDigest" + strconv.Itoa(i+1),
+			CPUTimeMs:  uint32(i + 1),
+		})
+	}
+	ts.Collect(timestamp, records)
+}
+
+func initializeCache(maxSQLNum int, addr string) *TopSQLCollector {
+	config := &TopSQLCollectorConfig{
+		PlanBinaryDecoder:   testPlanBinaryDecoderFunc,
+		MaxSQLNum:           maxSQLNum,
+		SendToAgentInterval: time.Minute,
+		AgentGRPCAddress:    addr,
```

Comment:
The agentGRPCAddress (and other configurations) are dynamic, and may be changed at any time. @crazycs520 PTAL